metaseq/cli/train.py

```diff
@@ -495,6 +495,57 @@ def post_checkpoint_callback(cfg, filename):
                 ),
             )
             os.remove(filename)
+
+            # Run evals after all checkpoint parts have been uploaded
+            checkpoint_id = checkpoint_dir.split("_")[-1]
+            if (
+                cfg.checkpoint.cloud_eval_script_path is not None
+                and distributed_utils.get_global_rank() == 0
+                and checkpoint_id.isdigit()
```

Comment:
This logic is brittle, could we rely on `num_updates` instead? Also we probably want evals at the end of the training run? See approach in https://github.com/facebookresearch/metaseq/pull/411/files

Reply:
- updated to use num_updates
- kept the logic for determining when to evaluate outside, as unlike your implementation its only for nfs
- I'm not sure we actually want to have an extra flag for evals at the end of training. Susan previously wanted me to remove similar logic